solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -99,4 +118,125 @@ impl FlashbotsApi {
+
+        match serde_json::from_str::<jsonrpc_core::Output>(&body) {
+            Ok(body) => match body {
+                jsonrpc_core::Output::Success(body) => {
+                    match serde_json::from_value::<FlashbotGasPrice>(body.result) {
+                        Ok(gas_price) => Ok(EstimatedGasPrice {
+                            eip1559: Some(GasPrice1559 {
+                                base_fee_per_gas: match u64::from_str_radix(
+                                    &gas_price.base_fee_per_gas[2..],
+                                    16,
+                                ) {
+                                    Ok(base_fee) => base_fee as f64,
+                                    Err(err) => {
+                                        return Err(anyhow!(
+                                            "failed to parse base_fee_per_gas: {}",
+                                            err
+                                        ))
+                                    }
+                                },
+                                max_fee_per_gas: match u64::from_str_radix(
+                                    &gas_price.default.max_fee_per_gas[2..],
+                                    16,
+                                ) {
+                                    Ok(max_fee_per_gas) => max_fee_per_gas as f64,
+                                    Err(err) => {
+                                        return Err(anyhow!(
+                                            "failed to parse max_fee_per_gas: {}",
+                                            err
+                                        ))
+                                    }
+                                },
+                                max_priority_fee_per_gas: match u64::from_str_radix(
+                                    &gas_price.default.max_priority_fee_per_gas[2..],
+                                    16,
+                                ) {
+                                    Ok(max_priority_fee_per_gas) => max_priority_fee_per_gas as f64,
+                                    Err(err) => {
+                                        return Err(anyhow!(
+                                            "failed to parse max_priority_fee_per_gas: {}",
+                                            err
+                                        ))
+                                    }
+                                },
+                            }),
+                            ..Default::default()
+                        }),
+                        Err(err) => Err(anyhow!("result not a FlashbotGasPrice: {}", err)),
+                    }
+                }
+                jsonrpc_core::Output::Failure(body) => Err(anyhow!(body.error)),
+            },
+            Err(err) => {
+                tracing::info!("flashbot cancellation response: {}", body);
+                Err(anyhow!(err))
+            }
+        }
+    }
```

Comment:
Tried to make this easier to read by eliminating right ward drift.

```suggestion
    pub async fn gas_price(&self) -> Result<EstimatedGasPrice> {
        let body = serde_json::json!({
            "jsonrpc": "2.0",
            "id": 1,
            "method": "eth_gasFees",
            "params": [],
        });
        let response = self.client.post(URL).json(&body).send().await?;
        let status = response.status();
        let body = response.text().await?;
        ensure!(status.is_success(), "status {}: {:?}", status, body);

        let output = serde_json::from_str::<jsonrpc_core::Output>(&body)
            .context(format!("body: {}", body))?;
        let success = match output {
            jsonrpc_core::Output::Success(success) => success,
            jsonrpc_core::Output::Failure(body) => anyhow::bail!(body.error),
        };
        let gas_price = serde_json::from_value::<FlashbotGasPrice>(success.result)
            .context("failed to deserialize FlashbotGasPrice")?;

        Ok(EstimatedGasPrice {
            eip1559: Some(GasPrice1559 {
                base_fee_per_gas: match u64::from_str_radix(&gas_price.base_fee_per_gas[2..], 16) {
                    Ok(base_fee) => base_fee as f64,
                    Err(err) => return Err(anyhow!("failed to parse base_fee_per_gas: {}", err)),
                },
                max_fee_per_gas: match u64::from_str_radix(
                    &gas_price.default.max_fee_per_gas[2..],
                    16,
                ) {
                    Ok(max_fee_per_gas) => max_fee_per_gas as f64,
                    Err(err) => return Err(anyhow!("failed to parse max_fee_per_gas: {}", err)),
                },
                max_priority_fee_per_gas: match u64::from_str_radix(
                    &gas_price.default.max_priority_fee_per_gas[2..],
                    16,
                ) {
                    Ok(max_priority_fee_per_gas) => max_priority_fee_per_gas as f64,
                    Err(err) => {
                        return Err(anyhow!("failed to parse max_priority_fee_per_gas: {}", err))
                    }
                },
            }),
            ..Default::default()
        })
    }
```

Reply:
Obsolete with the latest commit.